docusaurus/docs/ruby/reference.md

```diff
@@ -273,6 +273,30 @@ KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN="{spec/models/user_spec.rb,spec/controllers/
 
 Make sure to read the details in [Split by test examples](split-by-test-examples.mdx).
 
+## `KNAPSACK_PRO_SLOW_TEST_FILE_THRESHOLD`
+
+:::caution
+
+This option takes effect if you split test files by test cases among parallel CI nodes using:
+
+* [`KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES=true`](split-by-test-examples.mdx)
+
+:::
+
+Default: not set
+
+Available:
+
+- not set: automatically determines the slow test files threshold
+- Set a number of seconds. Test files above the threshold are split by test cases among parallel CI nodes.
+
+Example:
+
+```bash
+# 180 seconds (3 minutes)
+KNAPSACK_PRO_SLOW_TEST_FILE_THRESHOLD=180
+```
+
```

Comment:
I'd propose the following because it suggests it's best left to Knapsack and it's similar to what we already do for `KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN`.

---

## `KNAPSACK_PRO_SLOW_TEST_FILE_THRESHOLD` (Internal)

Set the time threshold above which test files are split by test cases (instead of letting Knapsack Pro decide for you).

Requires [`KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES=true`](split-by-test-examples.mdx).

This is supposed to be used by gem developers for debugging Knapsack Pro. But if you decide to use it, provide a big enough positive number. If the threshold is too low, it may match too many files and the test suite may slow down or fail: the Knapsack Pro API could time out, or CI could run out of memory.

Default: not set

Available: not set | seconds

Example:

```bash
# 180 seconds (3 minutes)
KNAPSACK_PRO_SLOW_TEST_FILE_THRESHOLD=180
```

Also, I see in the reference sometimes we say "not set" and some other times `nil`. I think we should always say "not set" because passing something like `KNAPSACK_PRO_TEST_FILE_LIST_SOURCE_FILE=nil` would not work. Prolly, "unset" would be even better than "not set" and it's shorter.

Reply:
Done:
* https://github.com/KnapsackPro/docs.knapsackpro.com/commit/d2da6c523070116fff01f41c2c72e3bd98c3c6bd
* https://github.com/KnapsackPro/docs.knapsackpro.com/commit/5aacb6c1ffa95a7da5fd133c93b6f251f0caf94c